lib/Payplug/Payplug.php

```diff
@@ -17,19 +19,47 @@ class Payplug
      */
     private $_token;
 
+    /**
+     * @var string The API Version.
+     */
+    private $_apiVersion;
+
     /**
      * Constructor for a authentication.
      *
-     * @param   string  $token  The token
+     * @param string $token The token
+     * @param string $api_version The API Version
      *
      * @throws Exception\ConfigurationException when token is not set
      */
-    public function __construct($token)
+    public function __construct($token, $apiVersion = null)
     {
         if (!is_string($token)) {
             throw new Exception\ConfigurationException('Expected string values for token.');
         }
         $this->_token = $token;
+        $this->_apiVersion = $apiVersion;
+    }
+
+    /**
+     * Initializes a Authentication and sets it as the new default global authentication.
+     * It also performs some checks before saving the authentication and set the API version
+     *
+     * @param array $props
+     * @return Payplug
+     * @throws ConfigurationException
+     */
+    public static function init($props = array())
```

Comment:
La valeur par défaut est contre-productive ici. Ça va planter deux lignes plus bas.

Reply:
Noté